Sources/RInAppMessaging/Extensions/UIApplication+IAM.swift

```diff
@@ -35,4 +35,30 @@ extension UIApplication {
         // Only iOS 13+ is supported
         return nil
     }
+    
+    private static let notificationSettingsURLString: String? = {
+        if #available(iOS 16, *) {
+            return UIApplication.openNotificationSettingsURLString
+        }
+        
+        if #available(iOS 15.4, *) {
+            return UIApplicationOpenNotificationSettingsURLString
+        }
+        
+        if #available(iOS 8.0, *) {
+            return UIApplication.openSettingsURLString
+        }
+        
+        return nil
+    }()
+    
+    private static let appNotificationSettingsURL = URL(
```

Comment:
It could be method property of   `func openAppNotificationSettings()`. No need of having static property so that you can do nil check for `notificationSettingsURLString` instead of assigning  empty string.